components/html-block/html-block.js

```diff
@@ -0,0 +1,271 @@
+					const html = mathjax.document(root, { InputJax, OutputJax });
+					html.render();
+					return html;
+				};
+
+				//
+				//  Now do the usual startup now that the extensions are in place
+				//
+				window.MathJax.startup.defaultReady();
+			}
+		}
+	};
+
+	return new Promise(resolve => {
+		if (!document.head.querySelector('#d2l-mathjax')) {
+			const script = document.createElement('script');
+			script.id = 'd2l-mathjax';
+			script.async = 'async';
+			script.onload = resolve;
+			script.src = 'https://s.brightspace.com/lib/mathjax/3.1.2/mml-chtml.js';
+			document.head.appendChild(script);
+		}
+	});
+
+};
+
+/**
+ * A component for displaying user-authored HTML.
+ */
+class HtmlBlock extends LitElement {
+
+	static get properties() {
+		return {
+			/**
+			 * The HTML to be rendered
+			 */
+			html: { type: String }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+				overflow-wrap: break-word;
+				overflow-x: auto;
+				overflow-y: hidden;
+				position: relative;
+				text-align: left;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			h1, h2, h3, h4, h5, h6, b, strong, b *, strong * {
+				font-weight: bold;
+			}
+			h1 {
+				font-size: 2em;
+				line-height: 37px;
+				margin: 21.43px 0;
+			}
+			h2 {
+				font-size: 1.5em;
+				line-height: 27px;
+				margin: 19.92px 0;
+			}
+			h3 {
+				font-size: 1.2em;
+				line-height: 23px;
+				margin: 18.72px 0;
+			}
+			h4 {
+				font-size: 1em;
+				line-height: 20px;
+				margin: 21.28px 0;
+			}
+			h5 {
+				font-size: 0.83em;
+				line-height: 16px;
+				margin: 22.13px 0;
+			}
+			h6 {
+				font-size: 0.67em;
+				line-height: 13px;
+				margin: 24.97px 0;
+			}
+			pre {
+				font-family: Monospace;
+				font-size: 13px;
+				margin: 13px 0;
+			}
+			p {
+				margin: 0.5em 0 1em 0;
+			}
+			ul, ol {
+				list-style-position: outside;
+				margin: 1em 0;
+				padding-left: 3em;
+			}
+			ul, ul[type="disc"] {
+				list-style-type: disc;
+			}
+			ul ul, ul ol,
+			ol ul, ol ol {
+				margin-bottom: 0;
+				margin-top: 0;
+			}
+			ul ul, ol ul, ul[type="circle"] {
+				list-style-type: circle;
+			}
+			ul ul ul, ul ol ul,
+			ol ul ul, ol ol ul,
+			ul[type="square"] {
+				list-style-type: square;
+			}
+			a,
+			a:visited,
+			a:link,
+			a:active {
+				color: var(--d2l-color-celestine);
+				cursor: pointer;
+				text-decoration: none;
+			}
+			a:hover,
+			a:focus {
+				color: var(--d2l-color-celestine-minus-1);
+				outline-width: 0;
+				text-decoration: underline;
+			}
+			:host([dir="rtl"]) {
+				text-align: right;
+			}
+			:host([dir="rtl"]) ul,
+			:host([dir="rtl"]) ol {
+				padding-left: 0;
+				padding-right: 3em;
+			}
+		`;
+	}
+
+	render() {
+	}
+
+	async updated(changedProperties) {
+		super.updated(changedProperties);
+
+		if (!changedProperties.has('html')) return;
+
+		if (this._hasMath()) {
+			if (!mathJaxLoaded) {
+				mathJaxLoaded = loadMathJax();
+			}
+			await mathJaxLoaded;
```

Comment:
If `loadMathjax()` just always resolved the Promise you could turn these 4 lines into: `await loadMathJax();`.

Reply:
Haha... yep.  Read this after I replied to your previous comment.  😄